input/postgres/data_directory.go

```diff
@@ -0,0 +1,19 @@
+package postgres
+
+import (
+	"github.com/pganalyze/collector/state"
+)
+
+// SetDataDirectory - Finds the data_directory in the list of settings and sets
+// the Server.Config.DataDirectory field to its value. Does nothing if the
+// setting is not found (e.g., due to permission issues).
+func SetDataDirectory(server *state.Server, settings []state.PostgresSetting) {
+	// N.B.: we're not taking a mutex here, because the data directory will not change location,
+	// so we only run this once at collector startup
+	for _, setting := range settings {
+		if setting.Name == "data_directory" && setting.CurrentValue.Valid {
+			server.Config.DataDirectory = setting.CurrentValue.String
```

Comment:
I wonder if we should leave the actual setting of things to the caller, like we do in `logs.ValidateLogCollectionConfig`, i.e. just return the data directory value here.

Reply:
Sure, makes sense. I think I originally modeled this after SetLogTimezone because I think we'd need a mutex here as well. Since we don't (`data_directory` is not going to change), I'll return a value here.